python/cuxfilter/charts/datashader/datashader.py

```diff
@@ -474,10 +389,17 @@ def heatmap(
         "spread",
         width,
         height,
-        title,
-        timeout,
+        tile_provider=None,
+        title=title,
+        timeout=timeout,
+        legend=legend,
+        legend_position=legend_position,
         **library_specific_params,
     )
+    # since it's built using the scatter api,
+    # changing the chart_type after the chart is created
+    p.chart_type = "heatmap"
```

Comment:
If `chart_type` being set by `Scatter`? If `chart_type` is some classification that can't be reliably derived automatically from the implementation alone (e.g. this result is made with a `Scatter but is a heatmap for whatever purpose `chart_type` is used for) then I'd suggest not setting it automatically at all, and just being explicit about the chart type everywhere it needs to be set in the lowercase methods here like `heatmap`. `line` etc. That way it's clear where the responsibility for setting this value lies and there  aren't overrides in some places  but not others (so then comments like this are also not needed)

Reply:
If it is a property *on* the result, then `heatmap` could pass it as a configuration to `Scatter(...)` to set in its initializer (but I'd just suggest being explicit about passing the value from "outside" Scatter)

Reply:
fixed this, all chart_types are now explicitly set outside the base classes